pkg/controller/migmigration/stage.go

```diff
@@ -560,33 +560,40 @@ func (t *Task) allStagePodsMatch() (report []string, err error) {
 		return
 	}
 
-	podDList := corev1.PodList{}
-
-	options := k8sclient.MatchingLabels(t.stagePodLabels())
-	err = dstClient.List(context.TODO(), options, &podDList)
-	if err != nil {
-		err = liberr.Wrap(err)
-		return
-	}
 	srcClient, err := t.getSourceClient()
 	if err != nil {
 		err = liberr.Wrap(err)
 		return
 	}
-	podSList := corev1.PodList{}
-	err = srcClient.List(context.TODO(), options, &podSList)
 
-	dPods := make(map[string]string)
+	for _, srcNamespace := range t.PlanResources.MigPlan.GetSourceNamespaces() {
+		podDList := corev1.PodList{}
+		podSList := corev1.PodList{}
+		options := k8sclient.MatchingLabels(t.stagePodLabels()).InNamespace(srcNamespace)
+		err = dstClient.List(context.TODO(), options, &podDList)
```

Comment:
is this cached or is this a client call against the API server? 

It might be better to do the bigger lists across all namespaces, and then filter them based on this SourceNamespaces before doing the comparison

Reply:
Listing stage pods across all namespaces might include stage pods that weren't destroyed for any reason and still present in the source cluster. And if that namespace is not included in migration, we still would have a warning that `x` stage pod is missing. This shouldn't be the case. So, I think listing pods limited to the current plan is a better way to go. But we certainly can keep the code as is to list all stage pods across all namespaces.